modules/calib3d/src/usac.hpp

```diff
@@ -0,0 +1,549 @@
+enum LocalOptimMethod {NullLO, InLORsc, ItLORsc, ItFLORsc, InItLORsc, InItFLORsc, GC, SIGMA};
+enum ScoreMethod {RANSAC, MSAC, LMS, MLESAC, MAGSAC};
+enum VerificationMethod { NullVerifier, SprtVerifier, TddVerifier };
+enum PolishingMethod { NonePolisher, LSQPolisher, GCPolisher };
+enum ErrorMetric {DIST_TO_LINE, SAMPSON_ERR, SGD_ERR, SYMM_REPR_ERR, FORW_REPR_ERR, RERPOJ};
+
+class Model : public Algorithm {
+public:
+    virtual bool isFundamental () const = 0;
+    virtual bool isHomography () const = 0;
+    virtual bool isEssential () const = 0;
+    virtual bool isPnP () const = 0;
+
+    // getters
+    virtual int getSampleSize () const = 0;
+    virtual bool resetRandomGenerator () const = 0;
+    virtual int getMaxNumHypothesisToTestBeforeRejection() const = 0;
+    virtual PolishingMethod getFinalPolisher () const = 0;
+    virtual LocalOptimMethod getLO () const = 0;
+
+    virtual ErrorMetric getError () const = 0;
+    virtual EstimationMethod getEstimator () const = 0;
+    virtual ScoreMethod getScore () const = 0;
+    virtual int getMaxIters () const = 0;
+    virtual double getConfidence () const = 0;
+    virtual bool isTimeLimit () const = 0;
+    virtual double getThreshold () const = 0;
+    virtual VerificationMethod getVerifier () const = 0;
+    virtual SamplingMethod getSampler () const = 0;
+    virtual int getMaxSampleSizeLO () const = 0;
+    virtual double getTimeForModelEstimation () const = 0;
+    virtual double getSPRTdelta () const = 0;
+    virtual double getSPRTepsilon () const = 0;
+    virtual double getSPRTavgNumModels () const = 0;
+    virtual NeighborSearchMethod getNeighborsSearch () const = 0;
+    virtual int getKNN () const = 0;
+    virtual int getCellSize () const = 0;
+    virtual bool isSampleLimit () const = 0;
+    virtual double getRelaxCoef () const = 0;
+    virtual int getMaxTimeMcs() const = 0;
+
+    virtual int getSamplerLengthPNAPSAC () const = 0;
+    virtual int getFinalLSQIterations () const = 0;
+    virtual int getDegreesOfFreedom () const = 0;
+    virtual double getSigmaQuantile () const = 0;
+    virtual double getUpperIncompleteOfSigmaQuantile () const = 0;
+    virtual double getLowerIncompleteOfSigmaQuantile () const = 0;
+    virtual double getC () const = 0;
+    virtual double getMaximumThreshold () const = 0;
+    virtual double getGraphCutSpatialCoherenceTerm () const = 0;
+    virtual int getLOSampleSize () const = 0;
+    virtual int getLOThresholdMultiplier() const = 0;
+    virtual int getLOIterativeSampleSize() const = 0;
+    virtual int getLOIterativeMaxIters() const = 0;
+    virtual int getLOInnerMaxIters() const = 0;
+    virtual int getMaxSampleSizeLOiterative () const = 0;
+
+    virtual const std::vector<int> &getGridCellNumber () const = 0;
+    virtual int getImage1Width () const = 0;
+    virtual int getImage1Height () const = 0;
+    virtual int getImage2Width () const = 0;
+    virtual int getImage2Height () const = 0;
```

Comment:
width/height => Size?

Reply:
agree. @ivashmak, please, replace each pair of methods width/height with one "getImage1Size", "getImage2Size".

Reply:
fixed